File: services/typescript/koa/src/routes/middleware/log.ts
import type { Context, Middleware, Next } from "koa";
import { getLogger } from "telemetry";

export const logMiddleware: Middleware = (ctx: Context, next: Next) => {
  let method = ctx.request.method;
  let path = ctx.request.path;

  getLogger().debug({ method, path }, "Incoming request");

  return next();
};

File: services/go/gin/routes/middleware/log.go
package middleware

import (
	"log/slog"

	"github.com/gin-gonic/gin"
)

func LogMiddleware() gin.HandlerFunc {
	return func(c *gin.Context) {
		slog.Debug("Incoming request", "method", c.Request.Method, "path", c.Request.URL.Path)

		c.Next()
	}
}
